crates/aptos/src/common/types.rs

```diff
@@ -99,3 +152,100 @@ pub struct EncodingOptions {
     #[clap(long, default_value = "hex")]
     pub encoding: EncodingType,
 }
+
+#[derive(Debug, Parser)]
+pub struct PrivateKeyInputOptions {
+    /// Private key input file name
+    #[clap(long, group = "key_input", parse(from_os_str))]
+    private_key_file: Option<PathBuf>,
+    /// Private key encoded in a type as shown in `encoding`
+    #[clap(long, group = "key_input")]
+    private_key: Option<String>,
+}
+
+impl PrivateKeyInputOptions {
+    pub fn extract_private_key(&self, encoding: EncodingType) -> Result<x25519::PrivateKey, Error> {
+        if let Some(ref file) = self.private_key_file {
+            encoding.load_key(file.as_path())
+        } else if let Some(ref key) = self.private_key {
+            let key = key.as_bytes().to_vec();
+            encoding.decode_key(key)
+        } else {
+            Err(Error::CommandArgumentError(
+                "One of ['--private-key', '--private-key-file'] must be used".to_string(),
+            ))
+        }
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct PublicKeyInputOptions {
+    /// Public key input file name.
+    #[clap(long, group = "key_input", parse(from_os_str))]
+    public_key_file: Option<PathBuf>,
+    /// Public key encoded in a type as shown in `encoding`
+    #[clap(long, group = "key_input")]
+    public_key: Option<String>,
+}
+
+impl PublicKeyInputOptions {
+    pub fn extract_public_key(&self, encoding: EncodingType) -> Result<x25519::PublicKey, Error> {
+        if let Some(ref file) = self.public_key_file {
+            encoding.load_key(file.as_path())
+        } else if let Some(ref key) = self.public_key {
+            let key = key.as_bytes().to_vec();
+            encoding.decode_key(key)
+        } else {
+            Err(Error::CommandArgumentError(
+                "One of ['--public-key', '--public-key-file'] must be used".to_string(),
+            ))
+        }
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct KeyInputOptions {
+    #[clap(flatten)]
+    private_key_options: PrivateKeyInputOptions,
+    #[clap(flatten)]
+    public_key_options: PublicKeyInputOptions,
```

Comment:
I don't understand, they're all exclusive.  This comes back to providing the public key on the command line rather than from a file.  File is cumbersome in automation